packages/react-reconciler/src/ReactFiberClassComponent.js

```diff
@@ -668,6 +668,11 @@ function mountClassInstance(
         workInProgress,
         instance,
       );
+
+      ReactStrictModeWarnings.recordLegacyContextWarning(
```

Comment:
Looks like we're not catching functional components here. They can also have `contextTypes` and should also be warned about.